hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/ByteArrayObjectDataInputTest.java

```diff
@@ -0,0 +1,641 @@
+package com.hazelcast.internal.serialization.impl;
+
+import com.hazelcast.internal.serialization.SerializationService;
+import com.hazelcast.nio.Bits;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.EOFException;
+import java.nio.ByteOrder;
+
+import static java.nio.ByteOrder.BIG_ENDIAN;
+import static java.nio.ByteOrder.LITTLE_ENDIAN;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+/**
+ * ByteArrayObjectDataInput Tester.
+ *
+ */
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class ByteArrayObjectDataInputTest {
+
+    private SerializationService mockSerializationService;
+
+    private ByteArrayObjectDataInput in;
+
+    private final static byte[] INIT_DATA = new byte[]{0, 1, 2, 3, 4, 5, 6, 7, 8, 9 };
+
+    @Before
+    public void before() throws Exception {
+        mockSerializationService = mock(SerializationService.class);
+        in = new ByteArrayObjectDataInput(INIT_DATA, mockSerializationService, ByteOrder.BIG_ENDIAN);
+    }
+
+    @After
+    public void after() throws Exception {
+        in.close();
+    }
+
+    @Test
+    public void testInit() throws Exception {
+        in.init(INIT_DATA, 2);
+        assertArrayEquals(INIT_DATA, in.data);
+        assertEquals(INIT_DATA.length, in.size);
+        assertEquals(2, in.pos);
+    }
+
+    @Test
+    public void testClear() throws Exception {
+        in.clear();
+        assertNull(in.data);
```

Comment:
Minor comment: I always place a line between the action and the asserts. This way it is immediately clear where the verification begins based on the visual structure. This is a recommended practice in a lot of testing literature and mentioned on our wiki.